### DISCUSSION TITLE
v1.3.46 Modern cameras RAW support, PSD Support.  EXIF read on most supported formats. AVIF, JXL, HEIF decoding optimizations. libwebp CVE fix.

### DISCUSSION THREAD

**Discussion by sylikc** [sylikc/jpegview#244] [General]
## 1.3.46

I wanted to make a special thanks to [qbnu](https://github.com/qbnu) for the new formats, along with optimizations and improvements to existing formats!

_(Minor version incremented to reflect the many changes between this and the previous release.  It's tempting to keep "waiting for the next feature to be done" before releasing...)_

### New features:
- Add support for more RAW image formats with LibRaw, ex: CR3 (ICC profiles are supported)
  PR #211, thanks [qbnu](https://github.com/qbnu)
- Add basic support for PSD (Adobe Photoshop Document)
  * Color modes supported: Grayscale, RGB, CMYK, Multichannel, Duotone, Lab
  * 8-bit depth PSDs only
  * Compression modes supported: Raw/Uncompressed, RLE
  (See PR for detailed supported/unsupported breakdown)
  PR #233, thanks [qbnu](https://github.com/qbnu)
- Add support for reading EXIF data from JXL, HEIF, AVIF, WebP and PNG files
  PR #213, thanks [qbnu](https://github.com/qbnu)
- Misc EXIF Metadata fixes:
  * GPS info parsed from RAW metadata
  * GPS coordinates in degrees+decimal minutes will now be converted to degrees+minutes+seconds
  * Altitudes below sea level will now show negative
  * Metadata will be shown for either RAW or JPEG, with RAW taking precedence
  PR #211, thanks [qbnu](https://github.com/qbnu)
- RAW metadata's acquisition date shown in window title
  PR #242, thanks [qbnu](https://github.com/qbnu)
- Add multithreading when decoding with libavif and reduced memory usage
  Up to ~20% decoding speed improvement
  PR #204, thanks [qbnu](https://github.com/qbnu)
- Optimize JXL and HEIF RGBA -> BGRA conversion
  Speed up ~25% for JXL and ~150% for HEIF
  PR #234, thanks [qbnu](https://github.com/qbnu)
- New INI file setting: `WindowBorderlessOnStartup`
  On startup, set JPEGView window to borderless mode (title bar hidden)
- New INI file setting: `WindowAlwaysOnTopOnStartup`
  On startup, set JPEGView window to be always on top

### Bugs fixed:
- Adjust animation frame times to appear closer to their actual values
  Accounts for time spent processing frame.  (Minimum frame time is 10ms due to technical limitations)
  PR #205, thanks [qbnu](https://github.com/qbnu)
Changed behavior:
- INI Option: `UseEmbeddedColorProfiles`=true can be used without losing Animated PNG support
  _ICC profiles are not supported for Animated PNG_
  PR #202, thanks [qbnu](https://github.com/qbnu)
- Renamed INI Option: `OverrideOriginalFileWithoutSaveDialog` -> `OverwriteOriginalFileWithoutSaveDialog`
- Renamed INI Option: `TrimWithoutPromptLosslessJPEG` -> `CropWithoutPromptLosslessJPEG`

### Other changes:
- Updated to library versions:
  * libavif 1.0.1
  * libde265 1.0.12
  * libjxl 5d20fbe
  * libpng 1.6.40
  * **libwebp 1.3.2 (Fixes CVE-2023-4863)**
  * zlib 1.3
  PR #235, thanks [qbnu](https://github.com/qbnu)
- Official scoop.sh support in bucket `extras/jpegview-fork`
  PR #223, thanks [Tim Martin](https://github.com/t-mart)

### Localization updates:
- New language: `sl` Slovenian, thanks Jadran Rudec
- New language: `tr` Turkish, thanks Kaya Zeren
- New language: `bg` Bulgarian, thanks Ivan Georgiev
- New translation: `sr` Serbian, thanks Ljuba Cirovic
- New translation: `hu` Hungarian, thanks József Ujvári
- New translation: `fi` Finnish, thanks Kari Eveli
- New translation: `sk` Slovak, thanks Marián Hikaník
- New translation: `ta` Tamil, thanks Nirmal G.
- Updated Japanese translation, thanks [maboroshin](https://github.com/maboroshin)
- Updated Korean translation, thanks [VenusGirl](https://github.com/VenusGirl)
- Updated Russian translation, thanks Dmitry Yerokhin
- Updated Chinese (Taiwan) translation, by converting chs -> cht
- Updated French translation, thanks Ronan Plantec
- Updated Greek translation, thanks Michalis
- Updated Polish translation, thanks Marcin Czerwien
- Rename INI option: `bel` -> `be` (official ISO 639-1 naming)
- Rename language: `Belorussian` to `Belarusian` (official naming)

## SHA-256 Checksums

Verify the release with these checksums on the release files:
```sha256sum
f91c1135dea06acbd3f22f13683433fa3da5a772a4cdb26c5dfe67b082447409 *JPEGView32_1.3.46.7z
842a864ad7f26054aa86aed556a56eea129a1f9d958e6f96e983df899a2d1a9a *JPEGView32_en-us_1.3.46.msi
d0184fff039e60296c726f000311a38ce29f6ef9bc50307c856cbc6b7f0f8b86 *JPEGView32_WinXP_1.3.46.zip
06855623d543edc829f29dbb85b19e7431a5c27bd4d18e131db75171fd107a16 *JPEGView64_1.3.46.7z
3108a85b5d408fa17027b25454f0ee7e7b450e680dbc1dab68d26d9099e18bf2 *JPEGView64_en-us_1.3.46.msi
f8bea0cec1f3430b4eb37f2dfe3190324bf6e353b5bd65ec5a34b40b4a9a43a9 *JPEGView_1.3.46.7z
ba71d5983b6c9e2dc55abbec0da53be54e67cf3251d6e9d2d8be82fe7d85bbe5 *JPEGView_1.3.46.zip
```

<hr /><em>This discussion was created from the release <a href='https://github.com/sylikc/jpegview/releases/tag/v1.3.46'>v1.3.46 Modern cameras RAW support, PSD Support.  EXIF read on most supported formats. AVIF, JXL, HEIF decoding optimizations. libwebp CVE fix.</a>.</em>

**Comment by gitsjd** [comment]
To make this a "go-to" viewer, I would highly suggest to add the minimum common function of adding a pen tool so someone can add simple drawing on an image, so one can "highlight" an area of an image by drawing an arrow/circle. This is a very common need, not to be confused with more advanced photo editing.
